extensions/sql-database-projects/src/dialogs/deployDatabaseQuickpick.ts

```diff
@@ -227,7 +227,7 @@ export async function launchCreateAzureServerQuickPick(project: Project, azureSq
 		title: constants.enterPassword(name),
 		ignoreFocusOut: true,
 		value: password,
-		validateInput: input => !utils.isValidSQLPassword(input) ? constants.invalidSQLPasswordMessage(name) : undefined,
+		validateInput: input => !utils.isValidSQLPassword(input, user, true) ? constants.invalidSQLPasswordMessage(name) : undefined,
```

Comment:
You should also switch the more info link provided in the error message

Reply:
sure, the problem is that page also doesn't say all four rules are needed, it says 3 is enough. I think the doc is wrong https://docs.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=azuresqldb-current

Reply:
This page is only for SQL box I believe. At least I'm not seeing any differences between the two versions. Where did you get that link from?

(also you shouldn't be including the en-us in the link)

Reply:
is has a azuresqldb as param "https://docs.microsoft.com/sql/relational-databases/security/password-policy?view=**azuresqldb-current**"

Reply:
https://docs.microsoft.com/sql/relational-databases/security/password-policy?view=azuresqldb-current

Reply:
hmm let me check. good point

Reply:
Yes, but that parameter isn't guaranteed to actually do anything. It's up to the page to be able to handle it

(although this might be a good thing to get PMs to verify if that's intended, because it seems like that should be what happens in this case)

Reply:
looks like policies can be applied by users so we cannot have them here. we can keep the default policy still but we cannot get the policy setup for tenant for force in UI. will close this PR for now until figure out a better way for this poblem

Reply:
Do you want to have this check for Azure at all then if you can't guarantee the policy they have is the same? Might be better to just let them try so we aren't blocking users unnecessarily...